ui/app/components/totp-edit.js

```diff
@@ -57,29 +64,45 @@ export default class TotpEdit extends Component {
 
   @action
   reset() {
-    this.model.unloadRecord();
-    this.successCallback(null);
+    this.args.model.unloadRecord();
+    this.transitionToRoute(SHOW_ROUTE, this.args.model.name);
   }
 
   @action
-  delete() {
-    this.persist('destroyRecord', () => {
+  async deleteKey() {
+    try {
+      await this.args.model.destroyRecord();
       this.transitionToRoute(LIST_ROOT_ROUTE);
-    });
-  }
-
-  @action
-  create(event) {
-    event.preventDefault();
-    const modelId = this.model.name;
-    // prevent from submitting if there's no key
-    // maybe do something fancier later
-    if (isBlank(modelId)) {
+    } catch (err) {
+      // err will display via model state
```

Comment:
I'm not sure this is true for `destroyMethod` - I'd have to dig in the docs. Older patterns in the codebase relied on the model's errors state but we've moved away from that. It's likely we should explicitly track `errorMessage` here and set accordingly.